src/asmcnc/comms/router_machine.py

```diff
@@ -1538,15 +1538,30 @@ def wait_for_movement_to_complete(dt):
 
 
     def set_jobstart_z(self):
-        self.s.write_command('G10 L20 P1 Z0')
+        self.set_datum(z=0)
         Clock.schedule_once(lambda dt: self.strobe_led_playlist("datum_has_been_set"), 0.2)
-        self.get_grbl_status()
 
     def set_standby_to_pos(self):
         self.s.write_command('G28.1')
         Clock.schedule_once(lambda dt: self.strobe_led_playlist("standby_pos_has_been_set"), 0.2)
 
-    
+    def set_datum(self, x=None, y=None, z=None, relative=False):
+        if relative:
+            datum_command = 'G10 L2'
+        else:
+            datum_command = 'G10 L20 P1'
+
+        if x is not None:
+            datum_command += " X" + str(x)
+        if y is not None:
+            datum_command += " Y" + str(y)
+        if z is not None:
+            datum_command += " Z" + str(z)
+
+        self.s.write_command(datum_command)
+        self.get_grbl_status()
```

Comment:
Might want to double check that this gives grbl enough time to change the datum & get the status - Boris may be able to advise. It may be worth sending these commands as a sequential stream so that the next command doesn't send until the oks have come back

Reply:
The old Z datum setter function did it this way, so it should work right? At least as far as I've tested it works fine. I tried sequential stream but it messes with the LED command that the existing functions use afterwards

Reply:
great, no worries :)